### ISSUE TITLE
Can't add `@ ecx` when editing the type of a structure

### ISSUE THREAD

**Issue by reduf** [Vector35/binaryninja-api#2994]
Binary Ninja Version: 3.0.3233 Personal, e250d0a3
Platform: Windows 8.1 Version 6.3 (Build 9600)

Hey, there seem to be an issue when editing the type of a structure containing functions. Consider the following structure.
```c
struct Sha1
{
     uint32_t dummy;
};
struct Sha1_VTable
{
    void (__stdcall* Reset)(struct Sha1* this);
    void (__stdcall* Update)(struct Sha1* this, uint8_t const* data, uint32_t length);
};
```

I can edit the type of the functions, individually and add the `@ ecx` "specifier" to both `this` and it work fine. But, if I try to add it when editing the VTable, then I get a syntax error.

So let's say I edit to both types individually, I get the following structure:
```c
struct Sha1_VTable
{
    void (__stdcall* Reset)(struct Sha1* this @ ecx);
    void (__stdcall* Update)(struct Sha1* this @ ecx, uint8_t const* data, uint32_t length);
};
```

At this point, I can't update the struct anymore, without removing the `@ ecx`, forcing me to re-add them manually on every functions.

Thanks,
Laurent

**Comment by psifertex** [comment]
Will be resolved once #3007 lands.

**Comment by reduf** [comment]
nice! Thanks a lot.

**Comment by CouleeApps** [comment]
Resolved now that clang is on dev, build 3.0.3413-dev. These should be usable in any place a function parameter is allowed.
